test_conformance/api/test_queue_array_properties.cpp

```diff
@@ -0,0 +1,352 @@
+            {
+                requested_bitfield = requested_properties[i + 1];
+            }
+            if (requested_properties[i] == CL_QUEUE_SIZE)
+            {
+                requested_size = requested_properties[i + 1];
+            }
+        }
+
+        error = verify_if_properties_supported(deviceID, requested_bitfield,
+                                               requested_size);
+        if (error == TEST_SKIPPED_ITSELF)
+        {
+            log_info("TC result: skipped\n");
+            return CL_SUCCESS;
+        }
+        test_error(error,
+                   "Checking which queue properites supported failed.\n");
+    }
+
+    // continue testing if supported user properties
+    error = create_queue_and_check_array_properties(context, deviceID, queue,
+                                                    test_case);
+    test_error(error, "Queue properties array verification result failed.\n");
+
+    for (int i = 0; i < num_elements; ++i)
+    {
+        buf[i] = i;
+    }
+
+    streams[0] =
+        clCreateBuffer(context, CL_MEM_READ_WRITE | CL_MEM_USE_HOST_PTR,
+                       num_elements * sizeof(int), buf.data(), &error);
+    test_error(error, "clCreateBuffer failed.");
+    streams[1] =
+        clCreateBuffer(context, CL_MEM_READ_WRITE | CL_MEM_ALLOC_HOST_PTR,
+                       num_elements * sizeof(int), NULL, &error);
+    test_error(error, "clCreateBuffer failed.");
+
+    error = clSetKernelArg(kernel, 0, sizeof(streams[0]), &streams[0]);
+    test_error(error, "clSetKernelArg failed.");
+
+    error = clSetKernelArg(kernel, 1, sizeof(streams[1]), &streams[1]);
+    test_error(error, "clSetKernelArg failed.");
+
+    error = clEnqueueNDRangeKernel(queue, kernel, 1, NULL, &num_elements, NULL,
+                                   0, NULL, &event);
+    test_error(error, "clEnqueueNDRangeKernel failed.");
+
+    error = clWaitForEvents(1, &event);
+    test_error(error, "clWaitForEvents failed.");
+
+    error = clEnqueueReadBuffer(queue, streams[1], CL_TRUE, 0, num_elements,
+                                buf.data(), 0, NULL, NULL);
+    test_error(error, "clEnqueueReadBuffer failed.");
+
+    for (int i = 0; i < num_elements; ++i)
+    {
+        if (buf[i] != i)
+        {
+            log_error("ERROR: Output results mismatch.");
+            return TEST_FAIL;
+        }
+    }
+    log_info("TC result: passed\n");
+    return TEST_PASS;
+}
+
+int test_queue_array_properties(cl_device_id deviceID, cl_context context,
+                                cl_command_queue queue, int num_elements)
+{
+    int error = CL_SUCCESS;
+    std::vector<test_queue_array_properties_data> test_cases;
+    std::string test_kernel = {
+        "__kernel void vec_cpy(__global int *src, __global int *dst)\n"
+        "{\n"
+        "    int  tid = get_global_id(0);\n"
+        "\n"
+        "    dst[tid] = src[tid];\n"
+        "\n"
+        "}\n"
+    };
+
```

Comment:
Note: This is a question/suggestion and not a requirement!

As-written, this test will compile a program and create new buffers for each test case.  Since each test cases uses the same program source and the same size buffers, would it make sense to create them here instead and reuse them for each test case?  I suspect most of the execution time for this test - which admittedly is probably pretty short - will be consumed by compiling kernels currently.

Reply:
I think that removing execution part will solve this issue as well